File: WoWSFT-Shared/src/main/kotlin/WoWSFT/model/gameparams/ship/component/torpedo/Launcher.java
package WoWSFT.model.gameparams.ship.component.torpedo;

import WoWSFT.config.WoWSFT;
import WoWSFT.model.gameparams.TypeInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class Launcher
{
    private List<String> ammoList;
    private double barrelDiameter;
    private boolean canRotate;
    private List<List<Double>> deadZone;
    private List<Double> horizSector;
    private long id;
    private String index;
    private List<Double> mainSector;
    private String name;
    private int numAmmos;
    private int numBarrels;
    private List<Double> position;
    private List<Double> rotationSpeed;
    private List<Double> shootSector;
    private double shotDelay;
    private boolean smallGun;
    private double timeBetweenShots;
    private double timeToChangeAngle;
    private double timeToChangeSpeed;
    private List<Double> torpedoAngles;
    private TypeInfo typeinfo;
    private List<Double> vertSector;
}

File: WoWSFT-Shared/src/main/kotlin/WoWSFT/model/gameparams/ship/component/torpedo/Launcher.kt
package WoWSFT.model.gameparams.ship.component.torpedo

import WoWSFT.config.WoWSFT
import WoWSFT.model.gameparams.TypeInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class Launcher
{
    var ammoList = listOf<String>()
    var barrelDiameter = 0.0
    var canRotate = false
    @JsonInclude
    var deadZone = listOf<List<Double>>()
    var horizSector = listOf<Double>()
    var id = 0L
    var index = ""
    var mainSector = listOf<Double>()
    var name = ""
    var numAmmos = 0
    var numBarrels = 0
    var position = listOf<Double>()
    var rotationSpeed = listOf<Double>()
    var shootSector = listOf<Double>()
    var shotDelay = 0.0
    var smallGun = false
    var timeBetweenShots = 0.0
    var timeToChangeAngle = 0.0
    var timeToChangeSpeed = 0.0
    var torpedoAngles = listOf<Double>()
    var typeinfo = TypeInfo()
    var vertSector = listOf<Double>()
}